hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/logical/ValuesReduceRule.java

```diff
@@ -104,6 +106,12 @@ protected void apply(
             Values values
     ) {
         PlanNodeSchema schema = OptUtils.schema(values.getRowType());
+        if (filter != null && hasParamsOrColumns(filter.getCondition())
+                || project != null && project.getProjects().stream().anyMatch(ExpressionUtil::hasParamsOrColumns)) {
+            // cannot apply the rule at planning time
```

Comment:
I guess it's going to be a problem once we introduce support for dynamic params?

Reply:
Yeah, but this rule has multiple issues that need to be fixed first.